airflow/contrib/hooks/aws_athena_hook.py

```diff
@@ -26,5 +26,4 @@
 warnings.warn(
     "This module is deprecated. Please use `airflow.providers.amazon.aws.hooks.athena`.",
     DeprecationWarning,
```

Comment:
Does this change show correct stacklevel in all cases?

i.e. importing Hook directly vs importing this hook via operators in a separate DAG file?

Reply:
> Does this change show correct stacklevel in all cases?

The "correct stacklevel" is the main problem here 😞 

I'm not sure if this change will give users anything helpful. Why? Because the stacklevel never points to line in dagfile. I tried changing it from 0 to ~21... and I get the either from module level, dagbag or airflow cli. Never from the problematic DAG. That's why I think what is proposed in #11960 is more helpful for users.